analyses/cnv-frequencies/01-cnv-frequencies.py

```diff
@@ -0,0 +1,249 @@
+	                         "Variant_Type", "Dataset", "Disease"]), merging_list).fillna("")
+	return(cnv_frequency_df)
+
+
+def get_annotations(cnv_frequency_df, efo_mondo_mapping_file, rmtl_mapping):
+	# insert annotation columns in the CNV frequency dataframe
+	cnv_frequency_df["RMTL"] = cnv_frequency_df.insert(1, "RMTL", "")
+	cnv_frequency_df["Gene_Full_Name"] = cnv_frequency_df.insert(3, "Gene_Full_Name", "")
+	cnv_frequency_df["Variant_Category"] = cnv_frequency_df.insert(5, "Variant_Category", "")
+	cnv_frequency_df["OncoKB_Category"] = cnv_frequency_df.insert(6, "OncoKB_Category", "")
+	cnv_frequency_df["EFO_ID"] = cnv_frequency_df.insert(7, "EFO_ID", "")
+	cnv_frequency_df["MONDO_ID"] = cnv_frequency_df.insert(8, "MONDO_ID", "")
+	cnv_frequency_df.fillna("", inplace=True)
+
+	# populate CNV frequency dataframe with full gene names
+	mg = mygene.MyGeneInfo()
+	ensembl_gene_list = list(cnv_frequency_df["Gene_Ensembl_ID"].unique())
+	for ensembl_gene in ensembl_gene_list:
+	    gene_name = mg.getgene(ensembl_gene, fields="name")
+	    if (gene_name is None) or ("name" not in gene_name):
+	        continue
+	    cnv_frequency_df.loc[
+	            cnv_frequency_df.Gene_Ensembl_ID == ensembl_gene, "Gene_Full_Name"] = gene_name["name"]
+
+	# populate CNV frequency dataframe with EFO and MONDO disease accessions              
+	row_index = -1
+	efo_mondo_df = pd.read_csv(efo_mondo_mapping_file, sep="\t") 
+	for cancer_group in efo_mondo_df["cancer_group"]:
+	    row_index += 1
+	    if cancer_group in list(cnv_frequency_df.Disease):
+		    cnv_frequency_df.loc[cnv_frequency_df.Disease == cancer_group , "EFO_ID"] = efo_mondo_df.at[row_index, "efo_code"]
+		    cnv_frequency_df.loc[cnv_frequency_df.Disease == cancer_group , "MONDO_ID"] = efo_mondo_df.at[row_index, "mondo_code"]                 
+	                    
+	# populate CNV frequency dataframe with EFO and MONDO disease accessions              
+	row_index = -1
+	ensg_rmtl_df = pd.read_csv(rmtl_mapping, sep="\t")
+	ensg_rmtl_df = ensg_rmtl_df[ensg_rmtl_df.rmtl.isin(["Relevant Molecular Target"])].reset_index()
+	for ensembl_gene in ensg_rmtl_df["ensg_id"]:
+		row_index += 1
+		if ensembl_gene in list(cnv_frequency_df.Gene_Ensembl_ID):
+			cnv_frequency_df.loc[cnv_frequency_df.Gene_Ensembl_ID == ensembl_gene , "RMTL"] = ensg_rmtl_df.at[row_index, "rmtl"]
+	cnv_frequency_df.fillna("", inplace=True)
+	return(cnv_frequency_df)
+
+
+def main():
+	# get input parameters	
+	args = read_parameters()
+
+	# call functions to compute CNV gene-level and add functional annotations
+	all_tumors_df = merge_histology_and_cnv_data(args.HISTOLOGY_FILE, args.CNV_FILE)
+	cancer_group_cohort_df = get_cancer_groups_and_cohorts(all_tumors_df)
+	cnv_frequency_df = compute_variant_frequencies(all_tumors_df, args.PRIMARY_TUMORS, args.RELAPSE_TUMORS, cancer_group_cohort_df)
+	cnv_frequency_df = get_annotations(cnv_frequency_df, args.EFO_MONDO, args.ENSG_RMTL)
```

Comment:
I wonder how long does this step take on your computer. It is taking > 30 minutes on mine. I guess it could be caused by the low pandas version on 0.25.3 on the Docker image, as it might not optimized for the procedures used in `get_annotations`.

The following output steps look OK. I will follow up if there is any issue after running this step.

Reply:
I think the mygene queries are the most time consuming steps. It takes about 1/3 second to query 1 gene. There are 24,861 unique genes, which would take about 2-3 hours. The runtime would be too long.

I will create a ENSG ID <-> gene full name mapping table and share with you at this thread.

Reply:
@logstar maybe we should make a gene annotation module in which we would add a file with ENSG--> full name, oncokb annotation, HUGO-->ENS, etc, which everyone can then use in their modules?

Reply:
@jharenza Sounds good. I will add ENSG -> full name mapping to the `input` directory of the `snv-frequency` module, so other modules can directly access. This is also on my to-do list, as I am concerned that future mygene.info updates will make our results irreproducible.

All other mappings are listed as following:

- Gene symbol -> OncoKB cancer gene and oncogene/TSG: `analyses/snv-frequencies/input/oncokb_cancer_gene_list.tsv`
- cancer_group -> EFO and MONDO: `data/efo-mondo-map.tsv`
- ENSG -> RMTL: `data/ensg-hugo-rmtl-v1-mapping.tsv`
- Gene symbol -> gene type: `analyses/fusion_filtering/references/genelistreference.txt`

Reply:
ok - can you actually make it a new module and you can ref that in your `snv-frequency module`? we will also add gtex-> Uberon in v7

I would like to get these PRs merged sooner than later and have iterative/staggered updates so they don't become so long as they are now

Reply:
@jharenza Sure. I will make a ticket and create a module for the annotations.

Got it. I will favor merging incomplete short PRs for future work.

Reply:
The annotation module issue is created at <https://github.com/PediatricOpenTargets/ticket-tracker/issues/112>.